magento/inventory
# Fixed- configurable product not back in stock after updating qty for the variant product by API/import

### Preconditions (*)
1. Magento 2.4.6-p1
2. Indexing mode should be "Update by Schedule"
### Description (*)
If the configurable product goes out of stock after shipment is created. and you update variant product stock by API/import then the configurable product stock status was not updated automatically. In the backend configurable product status shows in stock but when you visit the configurable product in the frontend then it is showing as out of stock. 
### Fixed Issues (if relevant)

1. Related issue: https://github.com/magento/inventory/issues/3358

### Manual testing scenarios (*)

1. Configurable product set up with one or more simple products
2. At least one simple product in stock and other should be out of stock
3. Place an order with the whole qty for config’s variant product which is in stock 
4. Create shipment for the order which you placed
5. Go to the frontend and see config’s product goes out of stock.
6. Add qty for config’s variant product by Api (use this endpoint rest/all/V1/products/{SKU}/stockItems/{ID})
7. Go to the frontend and see config’s product still out of stock.

### Additional information (*)

In “cataloginventory_stock_status” table configurable product status is not updated after the variant product is back in-stock.

![stock-status-img](https://github.com/magento/inventory/assets/59193951/4f815e42-4720-4b34-b360-755971056c7b)

### Questions or comments


### Contribution checklist (*)
